src/core/common/Point.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Point.h"
+
+std::ostream& operator<<(std::ostream &stream, const Point& point) {
+    return stream << "(" << point.x << ", " << point.y << ")";
+}
+
+Point Point::move(float offset_x, float offset_y) const {
+    return Point(x + offset_x, y + offset_y);
```

Comment:
Why is that function creating a new Point instead of changing its own "x" and "x"?

Reply:
Because that's the point of the function.